packages/web/src/electron.js

```diff
@@ -103,6 +119,84 @@ const reformatURL = url => {
   return `${scheme}://-/${path}`
 }
 
+const registerBuild = directory => {
+  const handler = async (request, cb) => {
+    const indexPath = path.join(directory, 'index.html')
+    const filePath = path.join(directory, new url.URL(request.url).pathname)
+
+    const cbArgs = { path: (await getPath(filePath)) || indexPath }
+    cb(cbArgs)
+  }
+
+  session.defaultSession.protocol.unregisterProtocol(scheme)
+  session.defaultSession.protocol.registerFileProtocol(scheme, handler, err => {
+    if (err) console.error(err)
+  })
+}
+
+const downloadWebUpdateAndNotify = async () => {
+  const newBuildPath = path.resolve(app.getAppPath(), `${buildName}.tar.gz`)
+  const webUpdateDir = path.resolve(app.getAppPath(), `web-update`)
+  const fileStream = fs.createWriteStream(newBuildPath)
+
+  // Fetch the build tar.gz file and write to file
+  const data = await fetch(
+    `https://s3.us-west-1.amazonaws.com/${s3Bucket}/${buildName}.tar.gz`
+  )
+  await new Promise((resolve, reject) => {
+    data.body.pipe(fileStream)
+    data.body.on('error', reject)
+    fileStream.on('finish', resolve)
+  })
+
+  // Create web-update directory and untar the build inside
+  if (!fs.existsSync(webUpdateDir)) fs.mkdirSync(webUpdateDir)
+  tar.x({
+    cwd: path.resolve(app.getAppPath(), `web-update`),
+    file: newBuildPath
+  })
+
+  // Notify the user of the update
+  mainWindow.webContents.send('webUpdateAvailable')
+}
+
+const checkForWebUpdate = () => {
+  const webUpdateInterval = setInterval(async () => {
+    const data = await fetch(
+      `https://s3.us-west-1.amazonaws.com/${s3Bucket}/package.json`
+    )
+    const packageJson = JSON.parse(await data.text())
+    const newVersion = packageJson.version
+
+    let currentVersion = appVersion
+
+    // Additional check for the version from the build package.json
+    // Needed after web updates because the local package.json version is not updated
+    if (
+      fs.existsSync(path.resolve(app.getAppPath(), `${buildName}/package.json`))
+    ) {
+      const buidlPackageJson = JSON.parse(
+        fs.readFileSync(
+          path.resolve(app.getAppPath(), `${buildName}/package.json`)
+        )
+      )
+
+      currentVersion = buidlPackageJson.version
+    }
+
+    // If there is a patch version update, download it and notify the user
+    // Minor version updates should be handled by the autoupdater
+    if (
+      semver.major(currentVersion) === semver.major(newVersion) &&
+      semver.minor(currentVersion) === semver.minor(newVersion) &&
+      semver.patch(currentVersion) < semver.patch(newVersion)
+    ) {
+      clearInterval(webUpdateInterval)
+      downloadWebUpdateAndNotify()
+    }
+  }, 10000)
```

Comment:
can we pull the 10,000 up to the top in a const called POLL_FOR_UPDATES_INTERVAL_MS? 60s seems fine too. 10 might be too agro